app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/integration/VpnNetworkStackProvider.kt

```diff
@@ -19,23 +19,26 @@ package com.duckduckgo.mobile.android.vpn.integration
 import com.duckduckgo.anvil.annotations.ContributesPluginPoint
 import com.duckduckgo.common.utils.plugins.PluginPoint
 import com.duckduckgo.di.scopes.VpnScope
+import com.duckduckgo.mobile.android.app.tracking.AppTrackingProtection
 import com.duckduckgo.mobile.android.vpn.AppTpVpnFeature
-import com.duckduckgo.mobile.android.vpn.VpnFeaturesRegistry
 import com.duckduckgo.mobile.android.vpn.network.VpnNetworkStack
 import com.squareup.anvil.annotations.ContributesBinding
 import javax.inject.Inject
 
 @ContributesBinding(VpnScope::class)
 class VpnNetworkStackProviderImpl @Inject constructor(
     private val vpnNetworkStacks: PluginPoint<VpnNetworkStack>,
-    private val vpnFeaturesRegistry: VpnFeaturesRegistry,
+    private val appTrackingProtection: AppTrackingProtection,
 ) : VpnNetworkStackProvider {
     override suspend fun provideNetworkStack(): VpnNetworkStack {
-        val features = vpnFeaturesRegistry.getRegisteredFeatures()
-        val feature = features.firstOrNull { it.featureName == AppTpVpnFeature.APPTP_VPN.featureName }
+        val networkStack = if (appTrackingProtection.isEnabled()) {
+            // if we have NetP enabled, return NetP's network stack
```

Comment:
This comment is not right, this provider returns either AppTP network if enabled or null.
I think we can remove this provider as it's always replaced by `NetpVpnNetworkStackProviderImpl`

Reply:
Ooh the comment was a copy paste error 😅 Deleted the class.